### PR TITLE
mat,all: Rename IsZero to IsEmpty

### LINKED ISSUE #1083: mat: Zero() is does not make IsZero() return true.
Right now we have `IsZero` which checks if the receiver is zero-sized, and `Zero`, which sets the matrix elements to zero. It is confusing that calling `Zero` does not make `IsZero` return true. I suggest we rename `IsZero` to `IsEmpty`. This removes the implicit connection, and `IsReset` seems linguistically awkward.

### PR DESCRIPTION
**Pull request by btracey**
This avoids the confusion between Zero() and IsZero() which sounds like they should be related
to one another but are not. This makes IsEmpty the counterpart to Reset. Add check for Zero in allMatrix

Fixes #1083.
Updates #1081.

Please take a look.

### CODE REVIEW

**Comment by btracey** [comment]
I went through mat and manually checked all of the instances of "zero" and "Zero" and made them consistent with the new usage (or at least tried). Outside of mat, I only fixed the places needed to compile. I'll go through the rest in fixing #1081

File: mat/matrix.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ type Matrix interface {
 // Dense types, especially helpful when adding new features.
 type allMatrix interface {
 	Reseter
-	IsZero() bool
+	IsEmpty() bool
+	Zero()
```
Comment (kortschak) [inline mat/matrix.go]: Why was this added?

Comment (btracey) [reply mat/matrix.go]: Because all of the types implement it and it's relevant to the update?

Comment (kortschak) [reply mat/matrix.go]: They all implement it, but it's not mentioned in the commit message.

Comment (btracey) [reply mat/matrix.go]: Done.

Comment (kortschak) [reply mat/matrix.go]: It seems not entirely related, but I'll let it go.

**Comment by btracey** [comment]
I thought it highlighted the confusion between Zero and IsZero

**Comment by ggggggggg** [comment]
This is a breaking change. Semantic versioning would require a version bump for this change.

**Comment by btracey** [comment]
Yes, sorry. We have https://github.com/gonum/gonum/issues/1089 to make this easier in the future, which will be done soon.

**Comment by kortschak** [comment]
@ggggggggg no. We are pre-1.0 and we note on the first line of the readme that we are unstable as shown by this badge [stability-unstable](https://github.com/emersion/stability-badges#unstable).

**Comment by ggggggggg** [comment]
I see. I misunderstood the import convention. I thought because ‘v1’ was in
the import path it was 1.0. I’m guessing this is part of the transition to
go modules, the docs say that if ‘v2’ is in the path is has to be version 2
but nothing for lower versions.

Sorry for the noise.

**Comment by kortschak** [comment]
No worries. The v1 actually predates modules and was our attempt to mitigate version problems in the future. It was presumptive and unfortunately a little temporally optimistic.
